packages/@okta/vuepress-site/use_cases/event_hooks/index.md

```diff
@@ -0,0 +1,139 @@
+
+ Delivery of events is best effort. Events are delivered at least once. Delivery may be delayed by network conditions. In some cases, multiple requests may arrive at the same time after a delay, or events may arrive out of order. To establish ordering, you can use the time stamp contained in the `data.events.published` property of each event.
+
+### Timeout and Retry
+
+When Okta calls your external service, it enforces a default timeout of 3 seconds. Okta will attempt at most one retry. Responses with a 4xx status code are not retried.
+
+### HTTP Headers
+
+The header of requests sent by Okta will look like this, provided you configure an authorization header, as recommended, and do not define additional custom headers:
+
+```http
+Accept: application/json
+Content-Type: application/json
+Authorization: ${key}
+```
+
+The value sent in the Authorization header is a secret string you provide to Okta when you register your event hook. This string serves as an API access key for your service, and Okta provides it in every request, allowing your code to check for its presence as a security measure. (This is not an Okta authorization token, it is simply a text string you decide on.)
+
+### Security
+
+To secure the communication channel between Okta and your external service, HTTPS is used for requests, and support is provided for header-based authentication. Okta recommends that you implement an authentication scheme using the authentication header, to be used to authenticate every request received by your external service.
+
+### Your Service's Responses to Event Delivery Requests
+
+Your external service's responses to Okta's ongoing event delivery POST requests should all be empty, and should have an HTTP status code of 200 (OK) or 204 (No Content).
+
+## Event Hook Setup
+
+### Registering an Event Hook
+
+After implementing your external service, you need to tell Okta it exists and configure it, by making a POST request to the `/api/v1/eventHooks` API. You use the JSON payload of that request to provide information on the event hook you are registering, including:
+
+- the URI of your external service's endpoint
+- the list of specific event types you want to use the event hook for
+- the secret value Okta should send in the authorization header of requests
+
+The response you receive will confirm creation of the event hook and provide you with the unique ID value of the created event hook.
+
+### Verifying an Event Hook
+
+After registering the event hook, you need to trigger the one-time verification process by making a POST request to the `/api/v1/eventHooks/${eventHookId}/verify` API. Okta will then make the GET request to your endpoint. If verification is successful, the JSON payload of the response to your call to the `/verify` API will contain a property called `verfication` set to `VERIFIED`.
+
```

Comment:
Recently, we introduced lifecycle APIs for an Event Hook (i.e. activate/deactivate) - https://oktainc.atlassian.net/browse/OKTA-202357

So, we need a small section called "Managing the lifecycle of an Event Hook". This section would appear after the section on "Triggering an Event Hook" and would contain the following sentences:

> 

- When a new Event Hook is created, it's status is ACTIVE by default. 
- One can deactivate an active Event Hook by making a POST request to the /api/v1/eventHooks/${eventHookId}/lifecycle/deactivate API. This would change the status of the event hook to INACTIVE. 
- In order to fire this inactive Event Hook, one needs to activate the hook again by making a POST request to the /api/v1/eventHooks/${eventHookId}/lifecycle/activate API. While this will change the Event Hook's status to ACTIVE again, however due to security reasons, Okta requires you to verify this newly reactivated event hook once again. Please follow the steps mentioned in "Verifying an Event Hook" on how to verify an event hook.

Reply:
We also need to mention that at a time an org can have a max number of 10 ACTIVE event hook configurations.

Reply:
I think it would make for explanation of lifecycle operations model for the management API in the reference page for the API.

Reply:
Sounds good. This is covered in Amogh's PR on the management API -- https://github.com/okta/okta-developer-docs/pull/193